src/gui/qgstemporallayerwidget.cpp

```diff
@@ -0,0 +1,194 @@
+{
+  QgsRasterLayer *layer = qobject_cast<QgsRasterLayer *>( mLayer );
+  QgsDateTimeRange range = layer->dataProvider()->temporalProperties()->fixedTemporalRange();
+  QgsDateTimeRange referenceRange = layer->dataProvider()->temporalProperties()->fixedReferenceTemporalRange();
+
+  if ( range.begin().isValid() && range.end().isValid() )
+  {
+    mStartTemporalDateTimeEdit->setDateTime( range.begin() );
+    mEndTemporalDateTimeEdit->setDateTime( range.end() );
+  }
+  if ( referenceRange.begin().isValid() && referenceRange.end().isValid() )
+  {
+    mStartReferenceDateTimeEdit->setDateTime( referenceRange.begin() );
+    mEndReferenceDateTimeEdit->setDateTime( referenceRange.end() );
+  }
+}
+
+void QgsTemporalLayerWidget::setInputWidgetState( QString type, bool enabled )
+{
+  if ( type == "normal" )
+  {
+    mStartTemporalDateTimeEdit->setEnabled( enabled );
+    mEndTemporalDateTimeEdit->setEnabled( enabled );
+    mSetEndAsStartNormalButton->setEnabled( enabled );
+  }
+  else if ( type == "reference" )
+  {
+    mStartReferenceDateTimeEdit->setEnabled( enabled );
+    mEndReferenceDateTimeEdit->setEnabled( enabled );
+    mSetEndAsStartReferenceButton->setEnabled( enabled );
+  }
+}
+
+void QgsTemporalLayerWidget::setMapCanvas( QgsMapCanvas *canvas )
+{
+  mCanvas = canvas;
+}
+
+void QgsTemporalLayerWidget::saveTemporalProperties()
+{
+  if ( mLayerRadioButton->isChecked() )
+  {
+    if ( mLayer->type() == QgsMapLayerType::RasterLayer &&
+         mLayer->providerType() == "wms" )
+    {
+      QgsDateTimeRange normalRange = QgsDateTimeRange( mStartTemporalDateTimeEdit->dateTime(),
+                                     mEndTemporalDateTimeEdit->dateTime() );
+      QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( mLayer );
+      rasterLayer->dataProvider()->temporalProperties()->setTemporalRange( normalRange );
+
+      if ( mDisableTime->isChecked() )
+        rasterLayer->dataProvider()->temporalProperties()->setEnableTime( false );
+      else
+        rasterLayer->dataProvider()->temporalProperties()->setEnableTime( true );
+
+      // Update current selection label
+      // updateRangeLabel( "layer", mRangeLabel );
+
+      if ( mReferenceCheckBox->isChecked() )
+      {
+        QgsDateTimeRange referenceRange = QgsDateTimeRange( mStartReferenceDateTimeEdit->dateTime(),
+                                          mEndReferenceDateTimeEdit->dateTime() );
+        rasterLayer->dataProvider()->temporalProperties()->setReferenceTemporalRange( referenceRange );
+        rasterLayer->dataProvider()->temporalProperties()->setReferenceEnable( true );
+      }
+      else
+        rasterLayer->dataProvider()->temporalProperties()->setReferenceEnable( false );
+    }
+
+    if ( mProjectRadioButton->isChecked() )
+    {
+      if ( mLayer->type() == QgsMapLayerType::RasterLayer &&
+           mLayer->providerType() == "wms" )
+      {
+        QgsDateTimeRange projectRange = mCanvas->mapSettings().temporalRange();
```

Comment:
This isn't right -- the canvas settings are only the CURRENT displayed temporal range, and will change on a frame-by-frame basis as the animation is played. Project settings would need to be stored elsewhere (in a QgsProjectTemporalSettings class)

Reply:
Thanks, For now I will just comment this line here, as I plan to work on this on a different PR.